### PR TITLE
Fix Issue 22340 - totalCPUs may not return accurate number of CPUs 

### PR DESCRIPTION
**Pull request by kubo39**

### CODE REVIEW

File: std/parallelism.d [hunk 1]
```diff
@@ -976,16 +976,47 @@ uint totalCPUsImpl() @nogc nothrow @trusted
...
+         */
+        for (int n = 64; n <= 16384; n *= 2)
         {
-            int count = CPU_COUNT(&set);
+            size_t size = CPU_ALLOC_SIZE(count);
+            if (size >= 0x400)
+            {
+                auto cpuset = cast(cpu_set_t*) calloc(1, size);
+                if (cpuset is null) break;
+                memset(cpuset, '\0', size);
```
Comment (MoonlightSentinel) [inline std/parallelism.d:1000]: `calloc` zeroes the allocated memory

Comment (kubo39) [reply std/parallelism.d:1000]: Ahh, thank you for the pointing out!

**Comment by thewilsonator** [comment]
Whats the status of this? Also does this have an issue number? If not please add one, or add a changelog entry.

**Comment by maxhaton** [comment]
@kubo39 SITREP on this? Would be good to get it in

**Comment by RazvanN7** [comment]
ping @kubo39

**Comment by RazvanN7** [comment]
@kubo39 This is still a draft. Is there anything more you would like to add?

**Comment by kubo39** [comment]
@RazvanN7 sorry for late reponse, I think its ready.

**Comment by RazvanN7** [comment]
Could you please provide a test case for this patch? Also, each bug fix should have an associated bugzilla entry.

**Comment by RazvanN7** [comment]
ping @kubo39

**Comment by ibuclaw** [comment]
> Could you please provide a test case for this patch? 

If only there was a CI provider who had a 1024 core machine. :-)

**Comment by RazvanN7** [comment]
@ibuclaw Do you think this is fine as is?

**Comment by RazvanN7** [comment]
@kubo39 I tried to add an empty commit with the message "Fix Issue 22340 - ..." however, I forgot to start a new branch and I accidentally overwritten your changes with an empty commit. Do you still have the local changes? If so, please be kind and push them again. Also, don't forget to link this issue: https://issues.dlang.org/show_bug.cgi?id=22340. Thanks!

**Comment by kubo39** [comment]
pushed and updated the title.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**